pkg/rtc/participant.go

```diff
@@ -842,7 +851,15 @@ func (p *ParticipantImpl) RemoveSubscribedTrack(subTrack types.SubscribedTrack)
 			numRemaining++
 		}
 	}
+
+	//
+	// NOTE
+	// subscribedTrackSettings should not deleted on removal as it is needed if corresponding publisher migrated
+	// LK-TODO: find a way to clean these up
+	//
+
```

Comment:
This is tricky. Leaving a TODO here for now. Concern is that this can grow unbounded. As remove could be getting called when unsubscribing from a remote before re-attaching to local (or vice-versa), cannot delete it on removal here. Have to pass around information about whether the publisher is actually leaving or migrating which gets cumbersome.

Reply:
I think we can clear it out here. If it's migrating out, the client will send up the latest information immediately following successful reconnect.